### PR TITLE
[FEATURE] : Horizontal bar chart 2way sync

### PR DESCRIPTION
**Pull request by DipraAich**
### Description
Merging horizontal and vertical bar chart visualizations
Implementing 2way sync in horizontal bar chart

### Issues Resolved
[#885](https://github.com/opensearch-project/observability/issues/885)

### Check List
- [ ] New functionality includes two-way sync for the Horizontal bar visualization.
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/visualizations/charts/__tests__/horizontal_bar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { Bar } from '../bar/bar';
+import {
+  LAYOUT_CONFIG,
+  HORIZONTAL_BAR_TEST_VISUALIZATIONS_DATA,
+} from '../../../../../test/event_analytics_constants';
+
+describe('Bar component', () => {
+  configure({ adapter: new Adapter() });
+
+  it('Renders bar component', async () => {
```
Comment (mengweieric) [inline dashboards-observability/public/components/visualizations/charts/__tests__/horizontal_bar.test.tsx:19]: Do we need to change it to corresponding name 'horizontal bar' in this case?

File: dashboards-observability/public/components/visualizations/charts/bar/bar.tsx [hunk 2]
```diff
@@ -48,7 +54,7 @@ export const Bar = ({ visualizations, layout, config }: any) => {
    * determine stylings
    */
   const barOrientation = dataConfig.chartStyles?.orientation || visMetaData.orientation;
-  const isVertical = barOrientation === visMetaData.orientation;
+  const isVertical = barOrientation === 'v';
```
Comment (mengweieric) [inline dashboards-observability/public/components/visualizations/charts/bar/bar.tsx:57]: Please use a constant variable for this, never use a direct string value.

File: dashboards-observability/public/components/visualizations/charts/helpers/viz_types.ts [hunk 3]
```diff
@@ -238,6 +239,14 @@ const getUserConfigs = (
   return isEmpty(configOfUser) ? userSelectedConfigs : configOfUser;
 };
 
+export const getVisTypeData = (vizId: string) => {
+  if (SIMILAR_VIZ_TYPES.includes(vizId)) {
+    return getVisType(vizId, { type: vizId });
+  } else {
```
Comment (mengweieric) [inline dashboards-observability/public/components/visualizations/charts/helpers/viz_types.ts:245]: We don't need else here just return 'getVisType(vizId);' as default.

File: dashboards-observability/public/components/visualizations/charts/__tests__/horizontal_bar.test.tsx (hunk 1 above)
Comment (DipraAich) [reply dashboards-observability/public/components/visualizations/charts/__tests__/horizontal_bar.test.tsx]: done

File: dashboards-observability/public/components/visualizations/charts/bar/bar.tsx (hunk 2 above)
Comment (DipraAich) [reply dashboards-observability/public/components/visualizations/charts/bar/bar.tsx]: done

File: dashboards-observability/public/components/visualizations/charts/helpers/viz_types.ts (hunk 3 above)
Comment (DipraAich) [reply dashboards-observability/public/components/visualizations/charts/helpers/viz_types.ts]: done

**Review by ps48 — APPROVED** [review]

**Review by mengweieric — APPROVED** [review]

**Review by kavithacm — APPROVED** [review]

**Pull request merged**